plugins/s3_auth/s3_auth.cc

```diff
@@ -440,6 +594,55 @@ str_concat(char *dst, size_t dst_len, const char *src, size_t src_len)
   return to_copy;
 }
 
+TSHttpStatus
+S3Request::authorize(S3Config *s3)
+{
+  TSHttpStatus status = TS_HTTP_STATUS_INTERNAL_SERVER_ERROR;
+  switch (s3->version()) {
```

Comment:
an enum is a good idea but, i'm okay with this.  It's pretty clear with version '2' and version '4'.